src/dnnl/dnnl_graph_compiler.hpp

```diff
@@ -0,0 +1,25 @@
+#ifndef GRAPH_COMPILER_HPP
+#define GRAPH_COMPILER_HPP
+
+#include "dnnl_graph_compiler.h"
+#include <string_view>
+
+struct dnnl_graph_compiler_executable {
+  // TODO: Implement
+};
+
+struct dnnl_graph_compiler {
+  const dnnl_graph_compiler_context ctx;
+
+  explicit dnnl_graph_compiler(const dnnl_graph_compiler_context *context)
+      : ctx(*context) {}
+
+  const dnnl_graph_compiler_executable *
+  compile(const std::string_view &graph_json) const;
+
+  void execute(const struct dnnl_graph_compiler_executable *exe,
```

Comment:
`execute` does not belong to the compiler.

Reply:
Moved to the executable class.